econml/sklearn_extensions/model_selection_utils.py

```diff
@@ -0,0 +1,434 @@
+        return {}
+        # raise ValueError("Invalid model type. Valid values are 'linear', 'forest', 'nnet', and 'poly'.")
+
+
+def select_regression_hyperparameters(estimator):
+    """
+    Returns a dictionary of hyperparameters to be searched over for a regression model.
+
+    Args:
+        model_type (str): The type of model to be used. Valid values are 'linear', 'forest', 'nnet', and 'poly'.
+
+    Returns:
+        A dictionary of hyperparameters to be searched over using a grid search.
+    """
+    if isinstance(estimator, ElasticNetCV):
+        return {
+            'l1_ratio': [0.1, 0.5, 0.9],
+            'max_iter': [1000],
+        }
+    elif isinstance(estimator, RandomForestRegressor):
+        return {
+            'n_estimators': [50],
+            'max_depth': [None, 10, 50],
+            'min_samples_split': [2, 5, 10],
+        }
+    elif isinstance(estimator, MLPRegressor):
+        # Hyperparameter grid for neural network classification model
+        return {
+            'hidden_layer_sizes': [(10,), (50,), (100,)],
+            'alpha': [0.0001, 0.001, 0.01],
+            'learning_rate': ['constant', 'adaptive']
+        }
+    elif isinstance(estimator, GradientBoostingRegressor):
+        # Hyperparameter grid for gradient boosting regression model
+        return {
+            'n_estimators': [100],
+            'learning_rate': [0.01, 0.1, 1.0],
+            'max_depth': [3, 5],
+        }
+    elif is_polynomial_pipeline(estimator=estimator):
+        # Hyperparameter grid for polynomial kernel classification model
+        return {
+            'linear__l1_ratio': [0.1, 0.5, 0.9],
+            'linear__max_iter': [1000],
+            'poly__degree': [2, 3, 4]
+        }
+    else:
+        warnings.warn("No hyperparameters for this type of model. There are default hyperparameters for ElasticNetCV, RandomForestRegressor, MLPRegressor, and the polynomial pipeline.", category=UserWarning)
+        return {}
+        # raise ValueError("Invalid model type. Valid values are 'linear', 'forest', 'nnet', and 'poly'.")
+
+
+def is_linear_model(estimator):
+    """
+    Check whether an estimator is a polynomial regression, logistic regression, linear SVM, or any other type of
+    linear model.
+
+    Args:
+    estimator (scikit-learn estimator): The estimator to check.
+
+    Returns:
+    is_linear (bool): True if the estimator is a linear model, False otherwise.
+    """
+
+    # Check if the estimator is a polynomial regression
+    if isinstance(estimator, Pipeline):
+        has_poly_feature_step = any(isinstance(step[1], PolynomialFeatures) for step in estimator.steps)
+        if has_poly_feature_step:
+            return True
+
+    # Check if the estimator is a linear regression or related model
+    if hasattr(estimator, 'fit_intercept') and hasattr(estimator, 'coef_'):
+        return True
```

Comment:
Again this seems like a roundabout test - I think checking for specific classes is probably more reasonable.